pkg/instancegroups/rollingupdate.go

```diff
@@ -69,6 +69,31 @@ type RollingUpdateCluster struct {
 	ValidateSuccessDuration time.Duration
 }
 
+// AdjustNeedUpdate adjusts the set of instances that need updating, using factors outside those known by the cloud implementation
+func (c *RollingUpdateCluster) AdjustNeedUpdate(groups map[string]*cloudinstances.CloudInstanceGroup, cluster *api.Cluster, instanceGroups *api.InstanceGroupList) error {
+	for _, group := range groups {
+		if group.Ready != nil {
+			var newReady []*cloudinstances.CloudInstanceGroupMember
+			for _, member := range group.Ready {
+				makeNotReady := false
+				if member.Node != nil && member.Node.Annotations != nil {
```

Comment:
(And FYI the same applies to nil maps, for reading - though not for writing, so again you don't need the Annotations != nil check)